src/vs/workbench/api/node/extHostMcpNode.ts

```diff
@@ -134,6 +138,42 @@ export class NodeExtHostMpcService extends ExtHostMcpService {
 	}
 }
 
+class McpHTTPHandleNode extends McpHTTPHandle {
+	protected override async _fetchInternal(url: string, init?: CommonRequestInit): Promise<Response> {
+		const { fetch, Agent } = await import('undici');
```

Comment:
I would pop this to

```
private readonly _undici = new Lazy(() => import('undici'))
```

and then use it as

```
// Note: imported async so that we can ensure we load undici after proxy patches have been applied
const { fetch, Agent } = this._undici.value;
```

to avoid going through the resolver multiple times.

Reply:
Done! Still needed an `await` but I think it will keep the existing resolved promise for the rest of the life of the `Lazy`, right?